Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -414,53 +414,91 @@ public DateTime Value
     private static TimeOnly? TruncateTimeToSeconds(DateTime dateTime) => new TimeOnly(dateTime.Hour, dateTime.Minute, dateTime.Second);
 
     /// <inheritdoc/>
-    public IDateTime ToTimeZone(string otherTzId)
-    {
-        if (IsFloating) return new CalDateTime(_dateOnly, _timeOnly, otherTzId);
+    IDateTime IDateTime.ToTimeZone(string otherTzId) => ToTimeZone(otherTzId);
 
-        var zonedOriginal = DateUtil.ToZonedDateTimeLeniently(Value, TzId);
-        var converted = zonedOriginal.WithZone(DateUtil.GetZone(otherTzId));
+    public CalDateTime ToTimeZone(string otherTzId)
```

Comment:
You asked whether `otherTzId` should be `string?`, and the answer was no.
But, well, sometimes it would be convenient to easily change a zoned date/time to Floating this way., WDYT - keep or change?

Reply:
As TzId `null` represents floating time everywhere throughout the class, it would be logical to allow `null` here. Otherwise we should introduce a method `ToFloating()`, but allowing null here keeps the API more straight forward. So I'd say, `null` should be allowed.

Reply:
Added support for null as I'm on it anyways.